sleuth/sleuth/enrich_tls.py

```diff
@@ -169,76 +107,87 @@ def tls_seclevel(scs, client_key_length, certs):
         data = json.load(f)
         params = data["tls_params"][scs]
 
-        kex = params['kex']
-        if kex == "RSA" or kex == "DH" or kex == "DHE_anon" or kex == "DHE_PSK" or kex == "SRP_SHA" or kex == "SRP_SHA_RSA":
-            if client_key_length < 1024:
-                kex_seclevel = 1 # avoid
-            elif client_key_length < 2048:
-                kex_seclevel = 2 # legacy
-            elif client_key_length < 3072:
-                kex_seclevel = 3 # acceptable
-            else:
-                kex_seclevel = 4 # recommended
-
-        elif kex == "ECDHE" or kex == "ECDH" or kex == "ECDH_anon":
-            if client_key_length < 224:
-                kex_seclevel = 1 # avoid
-            elif client_key_length < 256:
-                kex_seclevel = 2 # legacy
-            elif client_key_length < 512:
-                kex_seclevel = 3 # acceptable
-            else:
-                kex_seclevel = 4 # recommended
-        else:
-            kex_seclevel = kex_policy[kex]
+        policy_rules = policy.rules
+        min_seclevel = min(policy.classifications.values())
+        max_seclevel = max(policy.classifications.values())
 
+        # Analyze key exchange seclevel based on key length seclevel information
+        # provided in the policy json
+        #
+        kex = params['kex']
+        kex_policy = policy_rules["kex"]
+
+        kex_seclevel = UNKNOWN
+        if kex in kex_policy:
+            if client_key_length and "client_key_length" in kex_policy[kex]:
+                kex_seclevel = min_seclevel
+                for each in kex_policy[kex]["client_key_length"]:
+                    if client_key_length > int(each):
+                        kex_seclevel = kex_policy[kex]["client_key_length"][each]
```

Comment:
Slight error here. It will take the first value that the client_key_length is greater than. For instance, if the client_key_length is 512, but 224 is compared first, it will be assigned 2 when it really should be 4. Probably need some tracking mechanism similar to what you do below for the cert block.